include/kframework/foundry.md

```diff
@@ -246,7 +266,8 @@ First we have some helpers in K which can:
 ```k
     syntax KItem ::= #assume ( Bool ) [klabel(foundry_assume), symbol]
  // ------------------------------------------------------------------
-    rule <k> #assume(B) => . ... </k> ensures B
+    rule <k> #assume(B) ~> _ => #halt </k> <statusCode> _ => FOUNDRY_VACUOUSSUCCESS </statusCode> requires notBool B
+    rule <k> #assume(B) => . ... </k> requires B
```

Comment:
This will trigger a vacuous node being created on _every_ precondition that is inserted with `vm.assume`, won't it? So the user will be flooded with warnings about vacuous nodes, and we'll get extra branches in the KCFG.

It would be better if we detected the vacuous nodes directly using the backend, https://github.com/runtimeverification/haskell-backend/issues/3630

Or, in `pyk`, we can check every node we get back from the backend to see if it simplifies to `#Bottom`, and thus is vacuous.